drivers/flash/soc_flash_stm32f4.c

```diff
@@ -255,6 +254,12 @@ static struct flash_driver_api flash_stm32f4_api = {
 static int stm32f4_flash_init(struct device *dev)
 {
 	struct flash_priv *p = dev->driver_data;
+	int i;
+
+	for (i = 0; i < ARRAY_SIZE(sector); i++) {
+		sector[i].start = CONFIG_FLASH_BASE_ADDRESS +
+			i * sector[i - 1].len;
+	}
```

Comment:
if CONFIG_FLASH_BASE_ADDRESS can be any address in the flash space then I misunderstood the meaning of CONFIG_FLASH_BASE_ADDRESS. From what you say maybe it should be renamed to CONFIG_BASE_ADDRESS independently of where the address is located?